providers/reddit/reddit_test.go

```diff
@@ -0,0 +1,28 @@
+package reddit
+
+import (
+	"fmt"
+	"testing"
+)
+
+// TestBuildURI makes sure the BuildURI method builds valid URIs.
+func TestBuildURI(t *testing.T) {
+	p := Provider{}
+	cases := []struct {
+		query, want string
+	}{
+		// Regular reddit search.
+		{"best startups",
+			fmt.Sprintf("https://www.reddit.com/search?q=%s", "best+startups")},
+		// Subreddit search.
+		{"/r/cscareerquestions best startups",
```

Comment:
But I do like your idea, I already made the change. Take look regardless. It's cool because it actually gives meaningful result (well you can also say empty query in subreddit is meaningful I guess)